### PR TITLE
[CLEANUP beta] Drop export of built-ins, remove legacy components

### PR DESCRIPTION
**Pull request by mixonic**
Part of https://github.com/emberjs/ember.js/issues/19617

* Remove the export of `@ember/component/checkbox`
* Remove the export of `@ember/component/text-field`
* Remove the export of `@ember/component/text-area`
* Remove the export of `@ember/component/link-component`
* Remove the assignment of `Ember.Checkbox`
* Remove the assignment of `Ember.TextField`
* Remove the assignment of `Ember.TextArea`
* Remove the assignment of `Ember.LinkComponent`
* Remove the assignment of `Ember.TextSupport`
* Remove the assignment of `Ember.TargetActionSupport`
* ~Remove the assignment of `Ember._Input` (what is this?)~ I brought `_Input` back as it is under test. I'm not familiar with its usage.
* Hard-code `EMBER_MODERNIZED_BUILT_IN_COMPONENTS` to `true` path in most places. The remaining are a few spots where there are deprecations targeting *5.0*.
* Hard-code the built-in components (`<Textarea>`, `<Input>`, `<LinkTo>`) for the modernized path
* Remove the `query-params` helper (it is only needed by the legacy `LinkComponent` and was deprecated).
* Review https://github.com/emberjs/ember.js/pull/19669 and ensure all `disabledWhen` is removed.
* Review https://github.com/emberjs/ember.js/pull/19662 and ensure all `@tagName` support is removed
* Review all `ember.built-in-components.*` deprecations and ensure their functionality is removed.
* Remove code supporting `ember-glimmer.link-to.positional-arguments`, basically https://github.com/emberjs/ember.js/blob/master/packages/ember-template-compiler/lib/plugins/transform-link-to.ts

### CODE REVIEW

File: packages/@ember/-internals/glimmer/index.ts
Comment (mixonic) [inline packages/@ember/-internals/glimmer/index.ts:299]: New internal export used in tests that used to assert with the `LinkComponent`, but seem like they should test with this export now.

File: packages/@ember/-internals/glimmer/lib/components/link-to.ts
Comment (mixonic) [inline packages/@ember/-internals/glimmer/lib/components/link-to.ts:41]: Docs moved here from the deleted `-link-to.ts`

File: packages/@ember/-internals/glimmer/lib/component.ts
Comment (mixonic) [inline packages/@ember/-internals/glimmer/lib/component.ts:1090]: Per https://github.com/emberjs/ember.js/pull/19744#issuecomment-954956205 this deprecation (and the other in this file) were aimed at 4.x through the 3.x series, and that should not have been changed. A path should likely land *before* this PR lands fixing that in master and any releases impacted by the linked change.

**Comment by mixonic** [comment]
@chancancode this is ready for a close read if you can spare the time. I'm going to work next on:

* [ ] Resolving the bad deprecation change re: https://github.com/emberjs/ember.js/pull/19744#issuecomment-954956205. I want to land that before this.
* [ ] Trying to smoke test this with https://github.com/emberjs/ember-legacy-built-in-components/pull/1 in practice.

I'll leave a `DO NOT MERGE` flag until I've completed that validation.

File: packages/@ember/routing/index.ts
Comment (mixonic) [inline packages/@ember/routing/index.ts:1]: Per the RFC at https://emberjs.github.io/rfcs/0671-modernize-built-in-components-1.html, the `LinkTo` export here should be an opaque thing usable in strict mode. This should be updated to reflect that goal (it should re-export `LinkTo`).

The same for `Input` and `Textarea` should be audited.

Comment (mixonic) [reply packages/@ember/routing/index.ts:1]: Done.

File: packages/@ember/-internals/glimmer/lib/component.ts
Comment (mixonic) [reply packages/@ember/-internals/glimmer/lib/component.ts]: Done in https://github.com/emberjs/ember.js/pull/19823 and https://github.com/emberjs/ember.js/pull/19828

**Review by rwjblue — APPROVED** [review]

**Comment by ef4** [comment]
Nice lookin' PR.

**Pull request merged**
